pennylane/transforms/metric_tensor.py

```diff
@@ -387,6 +394,8 @@ def _metric_tensor_cov_matrix(tape, diag_approx):
         # for each operation in the layer, get the generator
         for op in curr_ops:
             gen, s = op.generator
+            if op.inverse:
+                s = -s

```

Comment:
Hi @dwierichs, I'm wondering: should we fix this behaviour instead by negating the scaling factor on the operation level?

I.e., a potential solution would be to make `generator` attribute a property:

For `qml.RX`, we'd go from

```python
generator = [PauliX, -1 / 2]
```
to
```python
@property
def generator(self):
    scaling_factor = -1 / 2 if not self.inverse 1 / 2
    return [PauliX, scaling_factor]
```
I think this would be the intuitive thing for a user, given the docstring of the `generator` attribute.

![Screenshot from 2021-12-06 12-10-10](https://user-images.githubusercontent.com/24476053/144890645-0a080da8-4276-4b13-8b25-7b282f766c04.png)

**Example**

If a user would like to naively apply the formula from the docstring, they could do so by gathering the gate parameter and the generator.
```python
op = qml.RY(0.3, wires=0).inv()

print('Theta: ', op.data, '\nInverse? ', op.inverse, '\nGenerator: ', op.generator)
```
```
Theta:  [0.3] 
Inverse?  True 
Generator:  [<class 'pennylane.ops.qubit.non_parametric_ops.PauliY'>, -0.5]
```
This is where the issue already arises: we store the information about taking the inverse of the operation in the `op.inverse` attribute.

Reply:
Note that this change would:
* Require changing the `generator` attribute on a case by case basis for several operations;
* Would likely affect the operation derivative capability ([see here](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/operation.py#L1833)) (and maybe other places too?);
* Potentially be revisited: having both `.inv()` and the `adjoint` method of an operation act separately seems to be causing strange situations (e.g., `inv()` only flips a flag, whereas `adjoint()` may negate the gate parameter).

Reply:
@antalszava the generator attribute is due to be refactored next week as part of the operator overhaul! So a hotfix for this  release should be okay

Reply:
I'm all in for deprecating the `inverse` flag :sweat_smile: I found it a bit counter-intuitive in all scenarios in which I came across it so far...
But I think it might be unnecessary to fix this now, given the changes that soon will be made to the Operator class. @mariaschuld what do you think about this? Would it pay off to address this issue now?

Reply:
Sounds good :+1: The changes here looked good as a hotfix.